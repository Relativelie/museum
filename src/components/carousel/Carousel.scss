@import "../../variables";


.carouselContent {
    position: relative;
    max-height: 650px;

    & .carouselContent_text {
        position: absolute;
        bottom: 159px;
        left: 50px;

        & h3 {
            @include titles($titleColor: $fontColorLight);
            margin: 0;
        }
    }

    & .carouselContent_img {
        width: 96%;
        min-height: 500px;
        object-fit: cover;
        z-index: -1;
    }
}

.slick-slider {
    margin-bottom: 70px !important;
}

.slick-slide div {
    display: flex !important;
    flex-direction: column;
    align-items: center;
}

// BUTTONS NEXT AND PREV
.slick-arrow {
    top: auto !important;
    bottom: -90px !important;
    width: clamp(40px, 6vw, 100px) !important;
    background-size: cover !important;
}

.slick-arrow::before {
    content: none !important;
}

.slick-prev {
    left: 42vw !important;
    background-position: center left;
    background-image: url("../../assets/images/carousel/prev.png") !important;
}

.slick-next {
    right: 42vw !important;
    background-position: center right;
    background-image: url("../../assets/images/carousel/next.png") !important;
}

@media all and (max-width: 800px) {
    .slick-arrow {
        width: 100px !important;
    }

    .slick-prev {
        left: 44px !important;
    }
    
    .slick-next {
        right: 44px !important;
    }
}