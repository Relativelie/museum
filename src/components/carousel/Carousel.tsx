export const Carousel = () => {
    return (
        <div className="carousel">
            <div>

            </div>
            <button>back</button>
            <button>next</button>
        </div>
    )
}