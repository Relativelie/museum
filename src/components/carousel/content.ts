import slideOne from '../../assets/images/carousel/img.png';

export const carouselContent = [
    {
        id: 1,
        name: 'Cлайд 4',
        image: slideOne,
    },
    {
        id: 2,
        name: 'Cлайд 2',
        image: slideOne,
    },
    {
        id: 3,
        name: 'Cлайд 3',
        image: slideOne,
    },
    {
        id: 4,
        name: 'Cлайд 4',
        image: slideOne,
    },
];
