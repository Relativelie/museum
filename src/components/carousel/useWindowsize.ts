import { useState, useEffect } from 'react';

export interface Size {
    width: number | undefined;
}

export const useWindowSize = (): Size => {
    const [windowSize, setWindowSize] = useState<Size>({
        width: undefined,
    });

    useEffect(() => {
        function handleResize() {
            setWindowSize({
                width: window.innerWidth,
            });
        }
        window.addEventListener('resize', handleResize);
        handleResize();
        return () => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};
