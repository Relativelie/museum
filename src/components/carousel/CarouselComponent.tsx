import { useEffect, useState } from "react";

import slideOne from "../../assets/images/carousel/img.png";
import Slider from "react-slick";
import "./Carousel.scss";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";



export const CarouselComponent = () => {
    const [translate, setTranslate] = useState(15);
    const [current, setCurrent] = useState(0);

    const test = [
        ["Cлайд 4", slideOne],
        ["Cлайд 1", slideOne],
        ["Cлайд 2", slideOne],
        ["Cлайд 3", slideOne],
    ]



    const settings = {
        className: "center",
        centerMode: true,
        infinite: true,
        slidesToShow: 1,
        speed: 500,
        initialSlide: 2
    };

    return (
        <Slider {...settings}>
            {test.map((elem, index) => {
                return (
                    <div className="carouselContent" key={`carousel-${index}`}>
                        <div className="carouselContent_text">
                            <h3>{elem[0]}</h3>
                        </div>
                        <img className="carouselContent_img" src={elem[1]} alt="slide picture" />
                    </div>
                )
            })}
        </Slider>
    );
}

