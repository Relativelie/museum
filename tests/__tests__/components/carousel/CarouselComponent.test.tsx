import { act, create } from 'react-test-renderer';
import { CarouselComponent } from '../../../../src/components/carousel/CarouselComponent';

describe('Carousel component', () => {
    test('snapshot - carousel component', () => {
        let root: any;
        act(() => {
            root = create(<CarouselComponent />);
        });
        expect(root.toJSON()).toMatchSnapshot();
    });
});
