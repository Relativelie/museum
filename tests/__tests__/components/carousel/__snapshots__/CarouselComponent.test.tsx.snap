// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component snapshot - carousel component 1`] = `
<div
  className="slick-slider center slick-initialized"
  dir="ltr"
>
  <button
    className="slick-arrow slick-prev"
    data-role="none"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
    type="button"
  >
     
    Previous
  </button>
  <div
    className="slick-list"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={null}
    onMouseMove={null}
    onMouseUp={[Function]}
    onTouchCancel={null}
    onTouchEnd={[Function]}
    onTouchMove={null}
    onTouchStart={[Function]}
    style={
      Object {
        "padding": "0px 50px",
      }
    }
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(300px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(300px)",
          "opacity": 1,
          "transform": "translate3d(300px, 0px, 0px)",
          "transition": "",
          "width": -1000,
        }
      }
    >
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={-2}
        onClick={[Function]}
        style={
          Object {
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 3
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={-1}
        onClick={[Function]}
        style={
          Object {
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 4
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-current"
        data-index={0}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 1
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={false}
        className="slick-slide slick-active slick-center"
        data-index={1}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 2
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide"
        data-index={2}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 3
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide"
        data-index={3}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 4
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={4}
        onClick={[Function]}
        style={
          Object {
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 1
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-center slick-cloned"
        data-index={5}
        onClick={[Function]}
        style={
          Object {
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 2
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={6}
        onClick={[Function]}
        style={
          Object {
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 3
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
      <div
        aria-hidden={true}
        className="slick-slide slick-cloned"
        data-index={7}
        onClick={[Function]}
        style={
          Object {
            "width": -100,
          }
        }
        tabIndex="-1"
      >
        <div>
          <div
            className="carouselContent"
            style={
              Object {
                "display": "inline-block",
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            <div
              className="carouselContent_text"
            >
              <h3>
                Cлайд 4
              </h3>
            </div>
            <img
              alt="slide"
              className="carouselContent_img"
              src="test-file-stub"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="slick-arrow slick-next"
    data-role="none"
    onClick={[Function]}
    style={
      Object {
        "display": "block",
      }
    }
    type="button"
  >
     
    Next
  </button>
</div>
`;
